wagtail/core/actions/revert_to_page_revision.py

```diff
@@ -0,0 +1,68 @@
+from django.core.exceptions import PermissionDenied
+
+
+class RevertToPageRevisionError(RuntimeError):
+    """
+    Raised when the revision revert cannot be performed for data reasons.
+    """
+
+    pass
+
+
+class RevertToPageRevisionPermissionError(PermissionDenied):
+    """
+    Raised when the revision revert cannot be performed due to insufficient permissions.
+    """
+
+    pass
+
+
+class RevertToPageRevisionAction:
+    def __init__(
+        self,
+        page,
+        revision,
+        user=None,
+        log_action="wagtail.revert",
+        submitted_for_moderation=False,
+        approved_go_live_at=None,
+        changed=True,
+        clean=True,
+    ):
+        self.page = page
+        self.revision = revision
+        self.user = user
+        self.log_action = log_action
+        self.submitted_for_moderation = submitted_for_moderation
+        self.approved_go_live_at = approved_go_live_at
+        self.changed = changed
+        self.clean = clean
+
+    def check(self, skip_permission_checks=False):
+        if self.page.alias_of_id:
+            raise RevertToPageRevisionError(
+                "Revisions are not required for alias pages as they are an exact copy of another page."
+            )
```

Comment:
Could we add another check in here to make sure that the given revision is for the given page?

Reply:
Users submit the revision id and the revision is taken from the following (`instance.revisions`):
```  
def _action_from_data(self, instance, data):
        revision = get_object_or_404(instance.revisions, id=data["revision_id"])

        return RevertToPageRevisionAction(
            page=instance, revision=revision, user=self.request.user
        )
 ```
 
 So if we have a revision when the action object is instantiated from the API, we can be sure that we have a valid revision for the given page...?

Reply:
Great, thanks! This looks good to me, sorry I missed that!